README.md

```diff
@@ -427,6 +427,74 @@ You can specify placeholder text for search- and select-type filters by setting
 
 If the filter is search-type, the `placeholder` attribute of the `<input>` element will be set to the placeholder value. If the filter is select-type, the text of the first `<option>` of the `<select>` element will be set to the placeholder value.
 
+#### Custom Filter Components
+
+You may want to render something other than a text box or select list for the filter. To do so, specify the component to render by providing
+the name in the `component` property of the `filter` object.
+
+For example, given that a component named "checkbox-filter" exists:
+
+```javascript
+[
+  {
+    key: 'name',
+    header: 'Name',
+    filter: {
+      component: 'checkbox-filter'
+    }
+  }
+]
+```
+
+Note that the `component` and `type` properties are mutually exclusive.
+
+The components will have access to the `columnDef` and `filter` properites. They contain the column definition and the filter value for
+that column, respectively. Setting the `filter` property in the component will invoke the `onReloadContent` action or cause the manual
+filter buttons to display based on the value of the `realtimeFiltering`.
+
+```handlebars
+{{!-- checkbox-filter.hbs --}}
+<div class="checkbox">
+  <label>
+    {{input type="checkbox" checked=filter}} Check if awesome
+  <label>
+</div>
+```
+
+Debouncing will be applied when the filter is set. If you don't want the delay, set the `debounce` property of the `filter` object to false. 
+
+```javascript
+[
+  {
+    key: 'name',
+    header: 'Name',
+    filter: {
+      component: 'checkbox-filter',
+      debounce: false
```

Comment:
I have a few thoughts on the new `debounce` property:

1. For optional object keys, I think it's preferable to make the default value "false" instead of "true" -- because then the "falsey" value of undefined is automatically equivalent to the default value. That might mean negating this property into something like `dontDebounce`.

2. Custom filters and non-custom filters now appear to have different configuration for debouncing (as in, this `debounce` property is being added for custom filters), and I think we should be careful to avoid having alternate configuration depending on whether the filter is a custom component or not. Or if I'm misunderstanding and it's added for any kind of filter, then we should document it outside of the custom filter component section of the README.

To take a step back, though, do you have a particular workflow in mind that requires column-by-column filter debounce configuration?